lib/ansible/modules/cloud/packet/packet_device.py

```diff
@@ -0,0 +1,568 @@
+HOSTNAME_RE = '({0}\.)*{0}$'.format(NAME_RE)
+MAX_DEVICES = 100
+
+PACKET_DEVICE_STATES = (
+    'queued',
+    'provisioning',
+    'failed',
+    'powering_on',
+    'active',
+    'powering_off',
+    'inactive',
+    'rebooting',
+)
+
+PACKET_API_TOKEN_ENV_VAR = "PACKET_API_TOKEN"
+
+
+ALLOWED_STATES = ['absent', 'active', 'inactive', 'rebooted', 'present']
+
+
+def serialize_device(device):
+    """
+    Standard represenation for a device as returned by various tasks::
+
+        {
+            'id': 'device_id'
+            'hostname': 'device_hostname',
+            'tags': [],
+            'locked': false,
+            'ip_addresses': [
+                {
+                    "address": "147.75.194.227",
+                    "address_family": 4,
+                    "public": true
+                },
+                {
+                    "address": "2604:1380:2:5200::3",
+                    "address_family": 6,
+                    "public": true
+                },
+                {
+                    "address": "10.100.11.129",
+                    "address_family": 4,
+                    "public": false
+                }
+            ],
+            "private_ipv4": "10.100.11.129",
+            "public_ipv4": "147.75.194.227",
+            "public_ipv6": "2604:1380:2:5200::3",
+        }
+
+    """
+    device_data = {}
+    device_data['id'] = device.id
+    device_data['hostname'] = device.hostname
+    device_data['tags'] = device.tags
+    device_data['locked'] = device.locked
+    device_data['ip_addresses'] = [
+        {
+            'address': addr_data['address'],
+            'address_family': addr_data['address_family'],
+            'public': addr_data['public'],
+        }
+        for addr_data in device.ip_addresses
+    ]
+    # Also include each IPs as a key for easier lookup in roles.
+    # Key names:
+    # - public_ipv4
+    # - public_ipv6
+    # - private_ipv4
+    # - private_ipv6 (if there is one)
+    for ipdata in device_data['ip_addresses']:
+        if ipdata['public']:
+            if ipdata['address_family'] == 6:
+                device_data['public_ipv6'] = ipdata['address']
+            elif ipdata['address_family'] == 4:
+                device_data['public_ipv4'] = ipdata['address']
+        elif not ipdata['public']:
+            if ipdata['address_family'] == 6:
+                # Packet doesn't give public ipv6 yet, but maybe one
+                # day they will
+                device_data['private_ipv6'] = ipdata['address']
+            elif ipdata['address_family'] == 4:
+                device_data['private_ipv4'] = ipdata['address']
+    return device_data
+
+
+def is_valid_hostname(hostname):
+    return re.match(HOSTNAME_RE, hostname) is not None
+
+
+def is_valid_uuid(myuuid):
+    try:
+        val = uuid.UUID(myuuid, version=4)
+    except ValueError:
+        return False
+    return str(val) == myuuid
+
+
+def listify_string_name_or_id(s):
```

Comment:
both callers of `listify_string_name_or_id` go on to call `.strip()` on list elements, so maybe you should get rid of that from here?